lib/puppet/provider/user/hpux.rb

```diff
@@ -19,14 +22,23 @@
     value !~ /\s/
   end
 
-  has_features :manages_homedir, :allows_duplicates, :manages_passwords
+  has_features :manages_homedir, :allows_duplicates, :manages_passwords, :manages_password_age, :manages_expiry
 
   def deletecmd
     super.insert(1,"-F")
   end
 
   def modifycmd(param,value)
-    super.insert(1,"-F")
+     cmd = super(param, value)
+     cmd << "-F"
+     if self.trusted == "Trusted"
+        cmd << ";"
```

Comment:
Question on mocking this for testing...

Instead of
        if provider.trust2
        resource.stubs(:allowdupe?).returns true
        provider.expects(:execute).with() { |args| 
args.include?('/usr/lbin/modprpw') and args.include?("-v") and 
args.include?("-l") }
        provider.uid = 1000

Would the correct be to do:

        let (:provider.trusted) { "Trusted" }
        resource.stubs(:allowdupe?).returns true
        provider.expects(:execute).with() { |args| 
args.include?('/usr/lbin/modprpw') and args.include?("-v") and 
args.include?("-l") }
        provider.uid = 1000

To test the below command works properly?

As said before I am learning so I am trying to figure this all out and 
clean things up where required.

I have not had much success inspecting the :provider / 
:resource.provider  to see what is contained within when I am trying to 
troubleshoot what is going on. It seems that the puppet wrapper around 
things is causing some difficulty getting useful debug output. I know 
this is just me learning Ruby, but it is frustrating all the same.


On 1/2/2015 2:15 PM, Michael Smith wrote:
>
> In lib/puppet/provider/user/hpux.rb 
> <https://github.com/puppetlabs/puppet/pull/3443#discussion-diff-22419246>:
>
> >
> >    def deletecmd
> >      super.insert(1,"-F")
> >    end
> >
> >    def modifycmd(param,value)
> > -    super.insert(1,"-F")
> > +     cmd = super(param, value)
> > +     cmd << "-F"
> > +     if self.trusted == "Trusted"
> > +        cmd << ";"
>
> What is this doing? Why use a semi-colon - which appears to just 
> ignore the 1st command - instead of just returning the modprpw call?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/puppetlabs/puppet/pull/3443/files#r22419246>.
>